select/index.js

```diff
@@ -0,0 +1,313 @@
+            </style>
+            <div class="irnmn-select ${this.selectedOption === null ? 'irnmn-select--unselected' : ''}" role="combobox" aria-expanded="${this.isOpen}" aria-haspopup="listbox" aria-labelledby="irnmn-select-header">
+                <div id="irnmn-select-header" class="irnmn-select__header" tabindex="0">
+                    ${this.selectedOption !== null ? this.options[this.selectedOption].name : this.headingText}
+                </div>
+                <ul id="option-list" class="irnmn-select__list ${this.isOpen ? 'irnmn-select__list--open' : ''}" role="listbox" aria-labelledby="irnmn-select-header">
+                    ${this.placeholder ? `
+                        <li class="irnmn-select__item irnmn-select__item--unselectable" role="option" aria-selected="false" tabindex="-1" disabled>
+                            ${this.placeholder}
+                        </li>` : ''}
+                    ${this.options
+                        .map(
+                            (option, index) => `
+                        <li class="irnmn-select__item" 
+                            role="option" 
+                            id="option-${index}" 
+                            tabindex="-1"
+                            aria-selected="${this.selectedOption === index ? 'true' : 'false'}"
+                            data-index="${index}"
+                            data-value="${option.value}">
+                            ${option.name}
+                        </li>
+                    `,
+                        )
+                        .join('')}
+                </ul>
+            </div>
+        `;
+    }
+
+    setupEventListeners() {
+        this.addEventListener('click', this.handleClick);
+        this.addEventListener('keydown', this.handleKeydown);
+        this.addEventListener('focusout', this.handleFocusOut);
+    }
+
+    removeEventListeners() {
+        this.removeEventListener('click', this.handleClick);
+        this.removeEventListener('keydown', this.handleKeydown);
+        this.removeEventListener('focusout', this.handleFocusOut);
+    }
+
+    handleFocusOut = (event) => {
+        // If the newly focused element (relatedTarget) is not within the custom select, close the list
+        if (!this.contains(event.relatedTarget)) {
+            this.closeList();
+        }
+    };
+
+    handleClick = (event) => {
+        const header = event.target.closest('.irnmn-select__header');
+        const item = event.target.closest('.irnmn-select__item');
+    
+        if (header) {
+            this.toggleList();
+            return;
+        }
+    
+        if (item && !item.classList.contains('irnmn-select__item--unselectable')) {
+            this.selectOption(parseInt(item.dataset.index, 10));
+            this.closeList();
+            return;
+        }
+    
+        if (!event.target.closest('.irnmn-select')) {
+            this.closeList();
+        }
+    };
+
+    handleKeydown = (event) => {
```

Comment:
Can you create an external file for the keyboard? Something like accessbility?

Reply:
@samuelsilvapt I tried to move this to it's own file but failed because it shares a lot of functions with the main class so it made it difficult. See the params involved below. I felt it complicated the code more.

`export function handleKeydown(event, options, toggleList, selectOption, closeList, focusItem) {}`